### ISSUE TITLE
Support for compiling for Qt > 5.9 

### ISSUE THREAD

**Issue by NZSmartie** [LibrePCB/LibrePCB#230]
I've successfully compiled LibrePCB using Qt 5.7. But, as of writing this; Qt 5.10 is available which is the toolchain I installed initially on Windows. 

When building for Qt 5.10 Mingw, the build fails with the following error:
```
error: specialization of 'template<class T>QtPrivate::QForeachContainer' class in different namespace [-fpermissive]
```

The source of the issue is https://github.com/LibrePCB/LibrePCB/blob/3de6686162b204ce7296847a8d29447ce35705d4/libs/librepcb/common/fileio/serializableobjectlist.h#L376-L394

There isn't much information about `QForeachContainer`  but the issue was about the namespace being different. 

My my proposed fix was to surround the block with the new `QtPrivte` namespace instead of prefixing QForeachContainer:

```c++
/*****************************************************************************************
 * Prevent from using SerializableObjectList in a foreach loop because it always would
 * create a deep copy of the list! You should use C++11 range based for loops instead.
 ****************************************************************************************/

#if (QT_VERSION > QT_VERSION_CHECK(5, 9, 0))
namespace QtPrivate {
#endif

template <typename T, typename P>
class QForeachContainer<librepcb::SerializableObjectList<T, P>> { public:
    ~QForeachContainer() = delete;
};
template <typename T, typename P>
class QForeachContainer<const librepcb::SerializableObjectList<T, P>> { public:
    ~QForeachContainer() = delete;
};


#if (QT_VERSION > QT_VERSION_CHECK(5, 9, 0))
} // namespace QtPrivate
#endif
```

Since i'm new to LibrePCB, and not sure of the scope of this change, can someone else verify this compiles with Qt > 5.9 and functionality isn't broken?

**Comment by NZSmartie** [comment]
Further investigation shows that this issue appears to be specific with Windows. as macOS (OSX) builds are already happening with Qt 5.10. I've yet to verify if this also happens on Debian builds

**Comment by NZSmartie** [comment]
I used docker to create a Qt 5.10 build environment and observed the same compile error.  mentioned above and resolved with the fix as well.

`Dockerfile`
```Dockerfile
FROM rabits/qt:5.10-desktop

LABEL description="LibrePCB debian build environment for Qt 5.10"

RUN sudo apt-get update -qq && sudo apt-get install -y \
    git \
    build-essential \
    libglu1-mesa-dev \
    openssl \
    zlib1g zlib1g-dev \
    && sudo rm -rf /var/lib/apt/lists/*
```

```bash
docker build -t nzsmartie/librepcb . 
mkdir build # assuming we're already in LibrePCB's source directory
docker run --rm -v "$(realpath  .):/home/user/src" -w "/home/user/src/build" -itd --name librepcb-build nzsmartie/librepcb
docker exec librepcb-build qmake -r ../
docker exec librepcb-build make -j8
```

**Comment by NZSmartie** [comment]
Fixed in #231
